secondarylib-feedback/src/main/scala/com/socrata/datacoordinator/secondary/feedback/FeedbackSecondary.scala

```diff
@@ -41,27 +34,36 @@ trait ComputationHandler[CT,CV, RCI <: RowComputeInfo[CV]] {
 
   /**
    * Perform the computation of the target column for each RowComputeInfo
-   * @return The RowComputeInfo's and resulting JValue's zipped with indexes of the rows
+   * @return The RowComputeInfo's and resulting values zipped with indexes of the rows
    * @note This should not throw any exception other than a [[ComputationFailure]] exception
    */
-  def compute(sources: Iterator[(RCI, Int)]): Iterator[((RCI, JValue), Int)]
+  def compute[RowHandle](sources: Map[RowHandle, Seq[RCI]]): Map[RowHandle, Map[UserColumnId, CV]]
 
 }
 
 /**
  * A FeedbackSecondary is a secondary that processes updates to source columns of computed columns
  * and "feeds back" those updates to data-coordinator via posting mutation scripts.
  */
-abstract class FeedbackSecondary[CT,CV, RCI <: RowComputeInfo[CV]] extends Secondary[CT,CV] {
+abstract class FeedbackSecondary[CT,CV] extends Secondary[CT,CV] {
 
-  val log = org.slf4j.LoggerFactory.getLogger(classOf[FeedbackSecondary[CT,CV,RCI]])
+  val log = org.slf4j.LoggerFactory.getLogger(classOf[FeedbackSecondary[CT,CV]])
 
   val httpClient: HttpClient
   def hostAndPort(instanceName: String): Option[(String, Int)]
   val baseBatchSize: Int
   val internalMutationScriptRetries: Int
   val mutationScriptRetries: Int
-  val computationHandler: ComputationHandler[CT, CV, RCI]
+  val computationHandlers: Seq[ComputationHandler[CT, CV]]
+  /**
+   * @return The `user` to specify in mutation scripts
+   */
+  val user: String
+
+  /**
+   * @return The number of times to retry the computation step via replaying later.
+   */
+  val retryLimit: Int
```

Comment:
Nope, because `FbSI` is also abstract, so the need to define them falls through to the individual secondaries.